R/ggMarginalHelpers.R

```diff
@@ -60,26 +60,34 @@ needsFlip <- function(marg, type) {
   topAndBoxP || rightAndNonBoxP
 }
 
-margPlotNoGeom <- function(data, type, scatPbuilt, marginGroupColour,
-                           marginGroupFill) {
+margPlotNoGeom <- function(data, type, scatPbuilt, groupColour, groupFill) {
   
   mapping <- ggplot2::aes(x = var)
   
-  haveMargMap <- marginGroupColour || marginGroupFill
+  haveMargMap <- groupColour || groupFill
   
   if (haveMargMap) {
+    
+    # Make sure user hasn't mapped a non-factor 
+    if (data[["group"]][1] < 0) { 
```

Comment:
What does this check do? What's in `data[["group"]][1]`?

Reply:
ggplot2 creates a positive integer vector column in `data` that corresponds to the groups. If no groups were created (e.g., b/c the user didn't map a var like colour or fill) then ggplot2 uses a vector of `-1L`'s instead:

https://github.com/tidyverse/ggplot2/blob/8778b48b37d8b7e41c0f4f213031fb47810e70aa/R/grouping.r#L3

We could check for `-1L` instead of `<0` but I don't think it matters. 

I've added a unit test in 
c9f5f1b to make sure that ggplot represents the group vector like this in future versions.

Reply:
I still don't understand how this checks that the grouped variable is a character/factor data and not numeric. But if it works, it works :) I'll test later tonight hopefully!

Reply:
Sorry, my explanation wasn't clear - `data[["group"]][1]` will be -1L when there are no actual group levels. You can step through the unit test that I wrote for this to see what I mean.